src/rgw/rgw_sync.cc

```diff
@@ -1758,16 +1764,11 @@ class RGWMetaSyncShardCR : public RGWCoroutine {
           tn->set_flag(RGW_SNS_FLAG_ACTIVE); /* actually have entries to sync */
           tn->log(20, SSTR("mdlog_marker=" << mdlog_marker << " sync_marker=" << sync_marker.marker));
           marker = max_marker;
-          yield call(new RGWReadMDLogEntriesCR(sync_env, mdlog, shard_id,
-                                               &max_marker, INCREMENTAL_MAX_ENTRIES,
-                                               &log_entries, &truncated));
-          if (retcode < 0) {
-            tn->log(10, SSTR("failed to list mdlog entries, retcode=" << retcode));
-            yield lease_cr->go_down();
-            drain_all();
-            *reset_backoff = false; // back off and try again later
-            return retcode;
-          }
+
+	  if (!log_entries.empty()) {
+	    max_marker = log_entries.back().id;
+	  }
+
           for (log_iter = log_entries.begin(); log_iter != log_entries.end() && !done_with_period; ++log_iter) {
```

Comment:
@bodley Can you sanity check this?

Reply:
For now reverted these changes (as it breaks sync over multiple periods) and needs thorough design to address properly. Will update it as and when it is ready. Request you to review remaining changes meanwhile. Thanks!